src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs

```diff
@@ -60,6 +62,19 @@ public void AddLineMappingFor(RazorIRNode node)
                 return;
             }
 
+            var imports = CodeDocument.GetImportSyntaxTrees();
+            if (imports != null)
+            {
+                for (var i = 0; i < imports.Count; i++)
+                {
+                    if (string.Equals(imports[i].Source.FileName, node.Source.Value.FilePath, StringComparison.OrdinalIgnoreCase))
```

Comment:
Do we even need the `CodeDocument` property anymore? IMO it's nice to have but I am okay either way.